### DISCUSSION TITLE
Puma::HttpParserError since upgrading to 4.2.0

### DISCUSSION THREAD

**Discussion by lifeofguenter** [mastodon/mastodon#27027] [Q&A] [answered]
Hi all,

I am seeing a bunch of:

> 2023-09-21 17:54:57 +0000 HTTP parse error, malformed request ("GET /users/lifeofguenter" - (-)): #<Puma::HttpParserError: Invalid HTTP format, parsing fails. Are you trying to open an SSL connection to a non-SSL Puma?>

since upgrading to 4.2.0

I do not necessarily notice anything not working though?

**Comment by renchap** [comment]
This is not expected.

Something is sending HTTPS requests to Puma, but only requests from your reverse proxy (nginx?) should reach Puma.

**Reply by lifeofguenter** [reply]
I run the official docker image with traefik in-front, and haproxy as lb.

haproxy communicates via https to traefik.

puma itself should not be accessible other than via traefik.

**Reply by ClearlyClaire** [reply]
I don't think this is related to SSL, I think the reverse-proxy is connecting to puma through plain HTTP, but somehow includes a header puma cannot parse.

**Reply by lifeofguenter** [reply, marked as answer]
I switch away from CF and it resolved the issue. I think just overall it is difficult to use a CDN as most do not properly support `Vary` as cache-key.

Using now Nginx as caching server and it is having better results.

Thanks :)
